### ISSUE TITLE
Really weird keyup behavior with remapped ctrl keys

### ISSUE THREAD

**Issue by chiarnglin** [randyrants/sharpkeys#192]
Firstly, thanks for a terrific utility. Really solves many of my issues running macOS onto Shadow (Windows streaming service for gaming). I've run into a really weird situation with the keyup events that I'm hoping you can help troubleshoot. 

Firstly, here's my setup on Sharpkeys: 
![image](https://user-images.githubusercontent.com/4613820/60233514-7d19dd80-9855-11e9-95eb-93623bd63b8c.png)
 
We'll run a simple test. I press "C", "X", "Z", "modifier key", then release them again IN THAT ORDER. (borrowing https://w3c.github.io/uievents/tools/key-event-viewer.html)

Shift key: 
![image](https://user-images.githubusercontent.com/4613820/60233588-c407d300-9855-11e9-92e2-898e8bb6269a.png)

Alt key: 
![image](https://user-images.githubusercontent.com/4613820/60233607-dc77ed80-9855-11e9-8b0e-cc4ae403d532.png)

Now Control key: 
![image](https://user-images.githubusercontent.com/4613820/60233621-ebf73680-9855-11e9-81e0-afb6b8992bf3.png)

When using Shift or Alt keys, I note that the keyup events happen exactly when I release each key in order. However, when using the Control key, the "keyup" event doesn't fire for C, X, or Z while Control is held down. However, releasing Control suddenly sends "keyup" for all 4 keys simultaneously. 

I cleared out the Sharpkeys settings and rebooted, the (real) Control key then works fine again. This only happens after I remap. Obviously this has implications for any games that allow multiple keys to be pressed simultaneously, so that's how I started investigating. Any ideas? 

Happy to do some more testing to help figure this one out. I'm really stumped.

**Comment by chiarnglin** [comment]
Just to prove I'm not going crazy, here is the same test on Control key, but with Sharpkeys settings deleted. 
![image](https://user-images.githubusercontent.com/4613820/60233793-bd2d9000-9856-11e9-9476-57066281255e.png)

**Comment by chiarnglin** [comment]
Another test that might be related: 
Keydown "X", tap the modifier key, then release "X". 

Using Shift: 
![image](https://user-images.githubusercontent.com/4613820/60233917-244b4480-9857-11e9-8a5d-c09ba824a3c9.png)

Now using Control: (?!?!?!?!)
![image](https://user-images.githubusercontent.com/4613820/60233950-33ca8d80-9857-11e9-9cfe-da58ec429e1a.png)

Releasing the Control key will send a "keyup event" for both itself and "X", even though the "X" key is still physically being pressed. Since it's still pressed, another keydown event will fire for "X", and finally when you physically release the "X" key, it sends another keyup event.

**Comment by randyrants** [comment]
Not sure what I'm looking at but bluntly put, all of this is outside of SharpKeys' control.  The app exposes a Registry key that Windows then uses to remap keys: my app does nothing but track key codes and writes them into the Registry correctly.  If the keys are acting weirdly before and after the remapping, it's how Windows is handling the key presses/releases.  HTH!

**Comment by chiarnglin** [comment]
Thanks! I decided to do another test. This time I swapped around the mapping between Z and Control, and was able to reproduce the problem with Z instead of Control. So it likely doesn't have anything to do with SharpKeys, but rather something to do with this particular key on the keyboard. We can consider this closed for now, at least that gives me the next direction to dig, even though I'm no closer to solving my mystery...
